### PR TITLE
Addition of `geom_turn()` and `theme_turnPlot()`

### PR DESCRIPTION
**Pull request by n400peanuts**
### Note: I didn't manage to get the exact sample data he was using in the blog, so the plot will look different.

```
library(tidyverse)


#### load d_sample.csv ####
local_elpaco_dir <- file.path("","Users", "evaviviani", "github", "benchmarking_elpaco")
d <- read.csv(file.path(local_elpaco_dir, "d_sample.csv"))

d_simple <- d |> select(begin, end, duration, participant, uid,
                        nwords, nchar, n, rank, freq, overlap,
                        priorby, FTO, overlapped, talk_all, talk_rel,
                        load, transitions)

#### add `participant_int` ####
d_simple$participant <- as.factor(d_simple$participant)
d_simple$participant_int <- as.numeric(d_simple$participant)

#### add `window_size` and `window_breaks` ####
extract_length <- 600000 # 10 min
window_size <- 60000 # 1 min
window_breaks <- as.integer(c(0:round(extract_length/window_size)) * window_size)

#### create `extract` dataset with begin0/end0/line cols ####
extract <- d_simple |>
  mutate(end = end - min(begin, na.rm = TRUE), # reset timestamps to start from 0
         begin = begin - min(begin, na.rm = TRUE),
         line = cut(begin, window_breaks, right = FALSE, labels = FALSE)) |>
  drop_na(line) |>
  group_by(line) |>
  mutate(begin0 = begin - min(begin, na.rm = TRUE), # reset timestamps to 0 for each new line
         end0 = end - min(begin, na.rm = TRUE)) |>
  ungroup()

#### create basic template for plotting named `plot.base` and use `talkr::theme_turnPlot()`  ####
plot.base <- extract[1:2000,] |>
  ggplot(aes(y = participant_int)) +
  talkr::theme_turnPlot() +
  ylab("") + xlab("time (s)") +
  viridis::scale_fill_viridis(option = "plasma", direction = 1, begin = 0.2, end = 0.8) +
  scale_y_reverse(breaks = seq(1, max(extract$line, 1)),
                  labels = seq(1, max(extract$line, 1))) +
  scale_x_continuous(limits = c(0, window_size),
                     breaks = seq(0, window_size, 10000),
                     label = seq(0, window_size / 1000, 10),
                     oob = scales::oob_keep)
plot.base

#### use `geom_turn()` ####
plot.base +
  ggtitle("Ten minutes of conversation",
          subtitle = "Points are interjections, time moves left-right and top-bottom") +
  talkr::geom_turn(aes(
    xmin = begin0,
    xmax = end0,
    ymin = line - 0.5 + participant_int / 3 - 0.2,
    ymax = line - 0.5 + participant_int / 3 + 0.2,
    xpoint = begin0 + 200,
    ypoint = line - 0.5 + participant_int / 3,
    fillpoint = rank)
  )


```

### CODE REVIEW

File: R/geom_turn.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#' Title
```
Comment (bvreede) [inline R/geom_turn.R:1]:
```suggestion
#' Show turn-taking in visualized conversations
```

**Review by bvreede — CHANGES_REQUESTED** [review]
FAB.

Some changes requested; please check the .gitignore, and remove the working_example from the PR.

I'm not certain about both `create_turn_plot()` and `create_turn_plot_modified()`; they are not functionality for this package, but they are intended to generate a test object to verify the behavior of geom_turn, correct?

If so, they should be moved to a file in `/tests`.

Mostly though:  🎉🎉🎉🎉

File: .gitignore [hunk 2]
```diff
@@ -2,5 +2,13 @@
...
+.Rbuildignore
 inst/doc
 *.rda
+*.Rbuildignore
+*sample_test.csv
+*.DS_Store
+*NAMESPACE
+*DESCRIPTION
+*test-get_uid_metadata.R
+*try.R
```
[the remark this answers is not in the document]
Comment (n400peanuts) [reply .gitignore:14]: I have pushed a lot of stupid stuff to this branch by mistake, among which my .gitignore. I am going to revert to the version of .gitignore which is on main which is correct! sorryyyy

File: R/create_turn_plot.R [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#' create turn plot as originally designed by mark
```
[the remark this answers is not in the document]
Comment (n400peanuts) [reply R/create_turn_plot.R]: yes, I have removed them now. They were useful to me to develop geom_turn() at the beginning, but not anymore.

File: working_example.R [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# code from https://ideophone.org/snailplot-rmd/
```
[the remark this answers is not in the document]
Comment (n400peanuts) [reply working_example.R]: Mark has suggested meeting together to get a minimal reproducible example of the type of plot he wants me to work on for geom_turn. So this means that the code in `working_example.R` is not useful any more and I am going to add in the benchmarking repo the new one when I meet with Mark.

File: R/geom_turn.R [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    size = 2,
+    shape = 21,
+    stroke = 1
+  ),
+
+  draw_panel = function(data, panel_params, coord, ...) {
+    rect_data <- transform(data, xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax)
+    point_data <- transform(data, x = xpoint, y = -ypoint, fill = fillpoint)
+
+    grid::grobTree(
```
[the remark this answers is not in the document]
Comment (n400peanuts) [reply R/geom_turn.R:54]: So this should be a dependency of ggplot2, as it is listed among the others on ggplot2's [cran](https://cran.r-project.org/web/packages/ggplot2/index.html) webpage

**Review by bvreede — APPROVED** [review]
Fab!! Thanks, this is a massive step forward :)

I'll approve now, but before you merge, please:
- Add parameter descriptions to geom_turn
- Run devtools::document() to update the .Rd files, and remove the .Rd files for deleted functions

Confirm that all is OK by running Check in your local Rstudio. This should not give any warnings (Notes are OK). At this point, there are several warnings still, which means the package cannot be installed.

🎉🎉

**Comment by n400peanuts** [comment]
> Fab!! Thanks, this is a massive step forward :)
> 
> I'll approve now, but before you merge, please:
> 
> * Add parameter descriptions to geom_turn
> * Run devtools::document() to update the .Rd files, and remove the .Rd files for deleted functions
> 
> Confirm that all is OK by running Check in your local Rstudio. This should not give any warnings (Notes are OK). At this point, there are several warnings still, which means the package cannot be installed.
> 
> 🎉🎉

Ehyy barb, thanks!!! sorry my PR is a mess 😢 
I have run `devtools::document()` and indeed this generated a new NAMESPACE (which exported the new function geom_turn and the new ggproto class GeomTurn), removed the documentation of removed R files and update new ones. I have also run checks and it seems working. Can you confirm?

**Pull request merged**